### ISSUE TITLE
cross-compile branch fail.

### ISSUE THREAD

**Issue by awsa2ron** [awslabs/amazon-kinesis-video-streams-producer-c#11]
Hi,
I have setup a new ubuntu18.04 docker and try to make the "cross-compile" branch. But failed.
I followed those steps:
```
# Cross-compilation to ARM, no tests are run
sudo apt-get install gcc-arm-linux-gnueabi
sudo apt-get install g++-arm-linux-gnueabi
sudo apt-get install binutils-arm-linux-gnueabi

#for Linux
export CC=arm-linux-gnueabi-gcc CXX=arm-linux-gnueabi-g++ AR=arm-linux-gnueabi-ar
mkdir build && cd build
cmake .. -DBUILD_TEST=TRUE -DBUILD_OPENSSL=TRUE \
         -DBUILD_OPENSSL_PLATFORM=linux-generic32 \
         -DBUILD_LIBCURL_HOST_PLATFORM=x86_64-unknown-linux-gnu \
         -DBUILD_LIBCURL_DESTINATION_PLATFORM=arm-unknown-linux-uclibcgnueabi

make
```
Here is the log:
```
Scanning dependencies of target kvsCommonCurl
[ 52%] Building C object CMakeFiles/kvsCommonCurl.dir/src/source/Common/Auth.c.o
In file included from /usr/include/openssl/sha.h:13:0,
                 from /home/aaron/producer-c/src/source/Common/Include_i.h:16,
                 from /home/aaron/producer-c/src/source/Common/Auth.c:5:
/usr/include/openssl/e_os2.h:13:11: fatal error: openssl/opensslconf.h: No such file or directory
 # include <openssl/opensslconf.h>
           ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
CMakeFiles/kvsCommonCurl.dir/build.make:62: recipe for target 'CMakeFiles/kvsCommonCurl.dir/src/source/Common/Auth.c.o' failed
make[2]: *** [CMakeFiles/kvsCommonCurl.dir/src/source/Common/Auth.c.o] Error 1
CMakeFiles/Makefile2:106: recipe for target 'CMakeFiles/kvsCommonCurl.dir/all' failed
make[1]: *** [CMakeFiles/kvsCommonCurl.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by MushMal** [comment]
Looks like the issue with cross-compiling the libCurl. We are working on it currently.

**Comment by awsa2ron** [comment]
@MushMal Thank you!
Where can I have information about how to compile kvs\curl\openssl\crypto individually?
I think that will helpful when we already have some libs(.so) in my linux or workaround this issue for now.

**Comment by MushMal** [comment]
Normally, you would CMake cross-compile libCurl which, in turn should cross-compile the OpenSSL which, in turn cross-compiles libCrypto.

I don't yet have more info but will update as soon as we get the cross-compile working

**Comment by awsa2ron** [comment]
@MushMal 
I found that "-DBUILD_LIBCURL_DESTINATION_PLATFORM" should be the same as the prefix of your cross compile toolchain. it is "arm-linux-gnueabi" in my case, cause I apt-get install gcc-arm-linux-gnueabi & g++-arm-linux-gnueabi.

I will close this PR and try ti simplify it in another one.
Thanks.

**Comment by chehefen** [comment]
fixed in #23
